### PR TITLE
virtio-serial: boot guest with multiple pcis with multiple ports

### PR DESCRIPTION
**Pull request by nanliu-r**
ID: 1804126

Depends on [avocado-vt/#2514](https://github.com/avocado-framework/avocado-vt/pull/2514)

Signed-off-by: Nana Liu <nanliu@redhat.com>

### CODE REVIEW

**Comment by nanliu-r** [comment]
Test result
(1/2) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M: PASS (93.31 s)
 (2/2) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M: PASS (133.10 s)

**Comment by nanliu-r** [comment]
@yanan-fu @xiagao please help to review this, thanks a lot.

**Comment by yanan-fu** [comment]
@nanliu-r  Almost same with `boot_1_N_virtserialports`, but different configuration. I think no need to add new script.

**Comment by nanliu-r** [comment]
> @nanliu-r Almost same with `boot_1_N_virtserialports`, but different configuration. I think no need to add new script.

thanks and update.

**Comment by nanliu-r** [comment]
Test pass:
JOB LOG    : /home/kar/workspace/job-results/job-2020-04-22T03.54-ea60241/job.log
 (1/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.1_N.q35: PASS (60.14 s)
 (2/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M.q35: PASS (86.77 s)
 (3/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.1_N.q35: PASS (116.25 s)
 (4/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M.q35: PASS (135.15 s)

**Comment by nanliu-r** [comment]
Test results:

(1/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.1_N.q35: PASS (96.36 s)
 (2/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M.q35: PASS (141.91 s)
 (3/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.1_N.q35: PASS (433.85 s)
 (4/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M.q35:PASS (796.81 s)

**Comment by yanan-fu** [comment]
@xiagao  Could you help review, thanks!

File: qemu/tests/cfg/virtio_serial_file_transfer.cfg [hunk 1]
```diff
@@ -30,14 +30,23 @@
...
                     file_sender = guest
         - boot_with_multiple_virtserialports:
+            type = boot_N_M_virtserialports
+            only unix_socket
+            filesize = 512
             variants:
                 - 1_N:
-                    only unix_socket
-                    type = boot_1_N_virtserialports
                     driver_name = vioser
```
Comment (xiagao) [inline qemu/tests/cfg/virtio_serial_file_transfer.cfg:38]: a little issue, driver_name is defined in Line15

Comment (nanliu-r) [reply qemu/tests/cfg/virtio_serial_file_transfer.cfg]: thanks and update.

**Comment by xiagao** [comment]
LGTM
Acked-by: Xiaoling Gao <xiagao@redhat.com>

Btw, nana ,could you provide the latest results?

**Comment by nanliu-r** [comment]
test results:
 (1/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.1_N.q35: PASS (101.18 s)
 (2/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.2.0.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M.q35: PASS (142.87 s)
 (3/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.1_N.q35: PASS (458.56 s)
 (4/4) Host_RHEL.m8.u2.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.Win2019.x86_64.io-github-autotest-qemu.virtio_serial_file_transfer.unix_socket.boot_with_multiple_virtserialports.N_M.q35: \
PASS (805.99 s)

**Comment by yanan-fu** [comment]
Thanks all!

**Pull request merged**